chromium_src/chrome/browser/ui/views/bubble/bubble_contents_wrapper.cc

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <utility>
+
+#include "chrome/browser/ui/views/bubble/bubble_contents_wrapper.h"
+#include "components/sessions/content/session_tab_helper.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_contents_delegate.h"
+#include "content/public/browser/web_contents_observer.h"
+
+// Trezor library opens a popup window to connect with a device.
+// The buble delegate doesnt allow to open popups and we use
+// the Browser window delegate to redirect opening new popup content
+// to the Browser delegate instead of default one.
+// In order to close all popups we also save tab ids of each opened popup window
+// and close all with the bubble together.
+#define RenderViewHostChanged                                                \
+  SetWalletDelegate(                                                         \
+      base::WeakPtr<content::WebContentsDelegate> browser_delegate) {        \
+    browser_delegate_ = std::move(browser_delegate);                         \
+  }                                                                          \
+  void BubbleContentsWrapper::AddNewContents(                                \
+      content::WebContents* source,                                          \
+      std::unique_ptr<content::WebContents> new_contents,                    \
+      const GURL& target_url, WindowOpenDisposition disposition,             \
+      const gfx::Rect& initial_rect, bool user_gesture, bool* was_blocked) { \
+    if (!browser_delegate_) {                                                \
```

Comment:
I don't understand, if the browser delegate is gone then the relevant tab/window is closed so why would we open a popup?

Reply:
this is our delegate that we set here https://github.com/brave/brave-core/pull/10902/files#diff-3661279516698c582f40547ddad91e92ebc753dfe8338f31919faeeedee2a742R21 only for `WalletPanelUI`, so if this delegate is not set I call default handler for this popup

Reply:
but why wouldn't it be set? And if we can open the popup without it then why bother to set it in the first place?

Reply:
I see now, I though this was a static method, but you're calling the no-op superclass method which serves no purpose.

Reply:
done